### ISSUE TITLE
Pinning and mut

### ISSUE THREAD

**Issue by chrsan** [dtolnay/cxx#583]
Hi!

I'm trying to refactor my Skia bindings to use cxx.
A `Surface` (`SkSurface`) has an associated `Canvas` (`SkCanvas`) that I'm trying to return.

Here's a sample with some stuff omitted:

```rust
unsafe extern "C++" {
    type Canvas;
    type Surface;

    fn canvas(self: Pin<&mut Surface>) -> Pin<&mut Canvas>;
}
```

When building I get: `&mut return type is not allowed unless there is a &mut argument`

Maybe I'm missing something obvious, but I can't find anything in docs.

Thanks,
Christer

**Comment by chrsan** [comment]
I think adding this:

```rust
if let Some(ref r) = efn.receiver {
    if r.pinned && r.mutable {
        return;
    }
}
```

before https://github.com/dtolnay/cxx/blob/83cbe894dcf6605ab3ad44ca851476b6e3e7dcf1/syntax/check.rs#L467 would work.

**Comment by dtolnay** [comment]
What you wrote should have worked. I've published a fix in 1.0.19.

**Comment by chrsan** [comment]
Great, and thanks a lot for an awesome library! Just ported my library to use cxx yesterday and got rid of a massive amount of handwritten C bindings and all of the unsafe Rust code.
